src/adapter/graphql/tests/tests/test_tasks.rs

```diff
@@ -8,15 +8,35 @@
 // by the Apache License, Version 2.0.
 
 use async_graphql::*;
+use chrono::Utc;
 use kamu_task_system::*;
-use kamu_task_system_inmem::*;
 use opendatafabric::DatasetID;
 
+mockall::mock! {
+    TaskScheduler {}
+    #[async_trait::async_trait]
+    impl TaskScheduler for TaskScheduler {
+        async fn get_task(&self, task_id: &TaskID) -> Result<TaskState, GetTaskError>;
+        fn list_tasks_by_dataset<'a>(&'a self, dataset_id: &DatasetID) -> TaskStateStream<'a>;
+        async fn create_task(&self, plan: LogicalPlan) -> Result<TaskState, CreateTaskError>;
+        async fn cancel_task(&self, task_id: &TaskID) -> Result<TaskState, CancelTaskError>;
+        async fn take(&self) -> Result<TaskID, TakeTaskError>;
+        async fn try_take(&self) -> Result<Option<TaskID>, TakeTaskError>;
+    }
+}
+
```

Comment:
@zaychenko-sergei Note that I started using `mockall` library here. It allows be to test GQL interface without linking any concrete implementations - not even "in-memory". I think we should eventually move to this model in all adapters, so that we are not linking them to impl crates even in test builds.

Reply:
I see, pretty typical mocking approach. Asserting expected interactions.